pono.cpp

```diff
@@ -46,18 +46,81 @@ using namespace pono;
 using namespace smt;
 using namespace std;
 
-
 ProverResult check_prop(PonoOptions pono_options,
-                        Property & p,
-                        const TransitionSystem & ts,
+                        Term & prop,
+                        TransitionSystem & ts,
                         const SmtSolver & s,
                         std::vector<UnorderedTermMap> & cex)
 {
-  logger.log(1, "Solving property: {}", p.name());
+  // get property name before it is rewritten
+  const string prop_name = ts.get_name(prop);
 
+  logger.log(1, "Solving property: {}", prop_name);
   logger.log(3, "INIT:\n{}", ts.init());
   logger.log(3, "TRANS:\n{}", ts.trans());
 
+  // modify the transition system and property based on options
+  if (!pono_options.clock_name_.empty()) {
+    Term clock_symbol = ts.lookup(pono_options.clock_name_);
+    toggle_clock(ts, clock_symbol);
+  }
+  if (!pono_options.reset_name_.empty()) {
+    std::string reset_name = pono_options.reset_name_;
+    bool negative_reset = false;
+    if (reset_name.at(0) == '~') {
+      reset_name = reset_name.substr(1, reset_name.length() - 1);
+      negative_reset = true;
+    }
+    Term reset_symbol = ts.lookup(reset_name);
+    if (negative_reset) {
+      SortKind sk = reset_symbol->get_sort()->get_sort_kind();
+      reset_symbol = (sk == BV) ? s->make_term(BVNot, reset_symbol)
+                                : s->make_term(Not, reset_symbol);
+    }
+    Term reset_done = add_reset_seq(ts, reset_symbol, pono_options.reset_bnd_);
+    // guard the property with reset_done
+    prop = ts.solver()->make_term(Implies, reset_done, prop);
+  }
+
+  if (pono_options.mod_init_prop_) {
+    prop = modify_init_and_prop(ts, prop);
+  }
+
+  if (pono_options.static_coi_) {
+    /* Compute the set of state/input variables related to the
+       bad-state property. Based on that information, rebuild the
+       transition relation of the transition system. */
+    StaticConeOfInfluence coi(ts, { prop }, pono_options.verbosity_);
+  }
+
+  if (pono_options.promote_inputvars_) {
+    // this is a bit tricky. because promote_inputvar
+    // modifies inputvars_, should copy the set first
+    UnorderedTermSet inputvars = ts.inputvars();
+    for (auto iv : inputvars) {
```

Comment:
can it be `auto const & iv`?

Reply:
Good point! Addressed with be917378730b7130df2ccc9959c8037134a79fe7.